File: Oxiyopal.Models/Cliente.cs
using Oxiyopal.Models.Base;
using Oxiyopal.Models.ViewModels;
using System;
using System.Collections.Generic;
using System.Text;

namespace Oxiyopal.Models
{
    public class Cliente : EntityBase<int>
    {
        #region Cliente Natural
        public string Nombre { get; set; }
        public string Apellido { get; set; }
        public string Documento { get; set; }
        public string Telefono { get; set; }
        public string Direccion { get; set; }
        public string Descripcion { get; set; }
        #endregion
        public bool EsEmpresa { get; set; }

        public int CiudadId { get; set; }
        public Ciudad Ciudad { get; set; }
        public IList<Ubicacion> Ubicaciones { get; set; }
        public IList<Factura> Facturas { get; set; }
    }
}
        

File: Oxiyopal.WebUI/ClientApp/src/app/models/Cliente.ts
import { EntityBase } from './Base/EntityBase';
import { Ciudad } from './Ciudad';
import { Ubicacion } from './Ubicacion';
import { Factura } from './Factura';

export interface Cliente extends EntityBase<number> {

    nombre: string;
    apellido: string;
    documento: string;
    telefono: string;
    direccion: string;
    descripcion: string;

    esEmpresa: boolean;

    ciudadId: number;
    ciudad: Ciudad;
    ubicaciones: Ubicacion[];
    facturas: Factura[];
}

